provision/filter_plugins/aws_sg_rules.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
```

Comment:
@robled : By default , ec2_group module takes the following format of input . 
```yaml name: example ec2 group
    ec2_group:
    name: example
    description: an example EC2 group
    vpc_id: 12345
    region: eu-west-1a
    aws_secret_key: SECRET
    aws_access_key: ACCESS
    rules:
      - proto: tcp
        from_port: 80
        to_port: 80
        cidr_ip: 0.0.0.0/0
      - proto: tcp
        from_port: 22
        to_port: 22
        cidr_ip: 10.0.0.0/8
      - proto: tcp
        from_port: 443
        to_port: 443
        group_id: amazon-elb/sg-87654321/amazon-elb-sg
      - proto: tcp
        from_port: 3306
        to_port: 3306
        group_id: 123412341234/sg-87654321/exact-name-of-sg
      - proto: udp
        from_port: 10050
        to_port: 10050
        cidr_ip: 10.0.0.0/8
      - proto: udp
        from_port: 10051
        to_port: 10051
        group_id: sg-12345678
      - proto: icmp
        from_port: 8 # icmp type, -1 = any type
        to_port:  -1 # icmp subtype, -1 = any subtype
        cidr_ip: 10.0.0.0/8
      - proto: all
        # the containing group name may be specified here
        group_name: example
    rules_egress:
      - proto: tcp
        from_port: 80
        to_port: 80
        cidr_ip: 0.0.0.0/0
        group_name: example-other
        # description to use if example-other needs to be created
        group_desc: other example EC2 group
```
In order to be consistent across , multiple cloud providers i suggested the format in linchpin topology.
[`https://github.com/CentOS-PaaS-SIG/linch-pin/pull/138/files/4bdaf8ab6e971b661aa442a914ee57f6eff1ad44#diff-bc64fad881cf1fd195da1393e776430d`](https://github.com/CentOS-PaaS-SIG/linch-pin/pull/138/files/4bdaf8ab6e971b661aa442a914ee57f6eff1ad44#diff-bc64fad881cf1fd195da1393e776430d)
In order to incorporate the translation between formats , the following filter is written. It filters out the rules based on rule types to distribute across attributes rules, rules_egress as for inbound and outbound rules respectively in the   provision/roles/aws/tasks/provision_aws_sg.yml playbook.